client/my-sites/customer-home/main.jsx

```diff
@@ -0,0 +1,254 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+import { some, isEmpty } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import CardHeading from 'components/card-heading';
+import EmptyContent from 'components/empty-content';
+import ChecklistWpcom from 'my-sites/checklist/main';
+import JetpackChecklist from 'my-sites/plans/current-plan/jetpack-checklist';
+import QueryJetpackPlugins from 'components/data/query-jetpack-plugins';
+import Main from 'components/main';
+import VerticalNav from 'components/vertical-nav';
+import VerticalNavItem from 'components/vertical-nav/item';
+import { preventWidows } from 'lib/formatting';
+import SidebarNavigation from 'my-sites/sidebar-navigation';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import {
+	isJetpackSite,
+	getCustomizerUrl,
+	canCurrentUserUseCustomerHome,
+} from 'state/sites/selectors';
+import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import DocumentHead from 'components/data/document-head';
+import getSiteChecklist from 'state/selectors/get-site-checklist';
+import QuerySiteChecklist from 'components/data/query-site-checklist';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const ActionBox = ( { action, iconSrc, label } ) => {
+	const buttonAction = 'function' === typeof action ? { onClick: action } : { href: action };
```

Comment:
I don't know if this is any better, but we could make this function a bit 'purer' by not mixing types on `action`

```
const ActionBox = ( { href, iconSrc, label, onClick } ) => {
	const buttonProps = { href, onClick };
    return (
		<div className="customer-home__box-action">
        <Button { ...buttonProps }>
       .....


							<ActionBox
								onClick={ () => page( `/themes/${ siteSlug }` ) }
								label={ translate( 'Change theme' ) }
								iconSrc="/calypso/images/customer-home/theme.svg"
							/>
							<ActionBox
								href={ customizeUrl }
								label={ translate( 'Edit menus' ) }
								iconSrc="/calypso/images/customer-home/menus.svg"
							/>

```

Reply:
Nice suggestion, thanks